### PR TITLE
settingsmenu: In list selection show option description

### PR DESCRIPTION
**Pull request by obligaron**
Master | PR
--- | ---
grafik | grafik

### CODE REVIEW

File: Source/DiabloUI/settingsmenu.cpp [hunk 1]
```diff
@@ -73,16 +73,24 @@ void GoBackOneMenuLevel()
 	shownMenu = ShownMenuType::Settings;
 }
 
+void UpdateDescription(OptionEntryBase *pOption)
+{
+	auto paragraphs = WordWrapString(pOption->GetDescription(), rectDescription.size.width, GameFont12, 1);
+	strncpy(optionDescription, paragraphs.c_str(), sizeof(optionDescription));
```
Comment (ephphatha) [inline Source/DiabloUI/settingsmenu.cpp:79]: GetDescription returns a translated string of potentially multibyte utf8 codepoints, might as well use CopyUtf8 here to be safe.
```suggestion
	CopyUtf8(optionDescription, paragraphs, sizeof(optionDescription));
```

Comment (obligaron) [reply Source/DiabloUI/settingsmenu.cpp]: Fixed 🙂

File: Source/DiabloUI/settingsmenu.cpp [hunk 2]
```diff
@@ -73,16 +74,24 @@ void GoBackOneMenuLevel()
 	shownMenu = ShownMenuType::Settings;
 }
 
+void UpdateDescription(OptionEntryBase *pOption)
+{
+	auto paragraphs = WordWrapString(pOption->GetDescription(), rectDescription.size.width, GameFont12, 1);
+	CopyUtf8(optionDescription, paragraphs, sizeof(optionDescription));
+}
```
Comment (ephphatha) [inline Source/DiabloUI/settingsmenu.cpp:81]: This function can take `const OptionEntryBase &option` as the argument. Polymorphic calls still work on references even if it was using a virtual method.

**Review by ephphatha — APPROVED** [review]
Functions fine, good to have context while changing a setting (especially for those of us with dodgy short term memory 😂 )

File: Source/DiabloUI/settingsmenu.cpp (hunk 2 above)
Comment (obligaron) [reply Source/DiabloUI/settingsmenu.cpp]: Changed 🙂

**Review by ephphatha — APPROVED** [review]

**Pull request merged**